lib/dialyzer/src/erl_types.erl

```diff
@@ -5071,6 +5076,9 @@ t_form_to_string({type, _Anno, 'fun', [{type, _, any}, Range]}) ->
 t_form_to_string({type, _Anno, 'fun', [{type, _, product, Domain}, Range]}) ->
   "fun((" ++ flat_join(t_form_to_string_list(Domain), ",") ++ ") -> "
     ++ t_form_to_string(Range) ++ ")";
+t_form_to_string({type, _Anno, 'fun', [{user_type, _, product, Domain}, Range]}) ->
```

Comment:
Ditto. We could add a guard clause and fold the bodies into one, but just by default went with rule of three.